### PR TITLE
Add MHRise HZV command

### PR DESCRIPTION
**Pull request by ricochhet**

### CODE REVIEW

File: src/main/kotlin/autocomplete/MhrMonster.kt [hunk 1]
```diff
@@ -0,0 +1,35 @@
+package autocomplete
+
+import com.kotlindiscord.kord.extensions.utils.FilterStrategy
+import com.kotlindiscord.kord.extensions.utils.suggestStringMap
+import dev.kord.core.entity.interaction.AutoCompleteInteraction
+import dev.kord.core.event.interaction.AutoCompleteInteractionCreateEvent
+import java.io.File
+
+object MhrAutoCache {
+    private const val dir = "./src/main/resources/source_files/MonsterDataImages/assets/mhr"
```
Comment (Yofou) [inline src/main/kotlin/autocomplete/MhrMonster.kt:10]: nitpick: can you use absolute paths here, ie `src/main/resources/source_files/MonsterDataImages/assets/mhr`, both should work, but `./` can make it look like this path is relative to this file when it's not.

File: src/main/kotlin/autocomplete/MhrMonster.kt [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+    private val folder = File("$dir/monster/")
+    val monsters = folder.walkBottomUp().maxDepth(1).toSet().map { 
+        val name = File(it.toURI()).nameWithoutExtension
+        name.replace("_HZV", "") // Not sure if this'll actually work.
```
Comment (Yofou) [inline src/main/kotlin/autocomplete/MhrMonster.kt:14]: this does work, please remove the comment.

File: src/main/kotlin/extensions/mhr/Hzv.kt [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+fun String.capitalizeDashes(locale: Locale? = null): String {
+    return split("-").joinToString("-") { word ->
+        word.replaceFirstChar {
+            if (it.isLowerCase()) {
+                it.titlecase(locale ?: Locale.getDefault())
+            } else {
+                it.toString()
+            } }
+    }
+}
```
Comment (Yofou) [inline src/main/kotlin/extensions/mhr/Hzv.kt:26]: Do we need to handle this case?

File: src/main/kotlin/extensions/mhr/Hzv.kt [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+
+@KordPreview
+val MhrHzvCommand: suspend PublicSlashCommand<out MhrHzv>.() -> Unit = {
+    name = "hzv"
+    description = "Get hzv info for a specific monster\n"
+
+    val dir = "./src/main/resources/source_files/MonsterDataImages/assets/mhr"
```
Comment (Yofou) [inline src/main/kotlin/extensions/mhr/Hzv.kt:33]: nitpick: same comment as above `src/main/resources/source_files/MonsterDataImages/assets/mhr
"`

File: src/main/kotlin/extensions/mhr/Hzv.kt [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+    action {
+        val log = Logger.getLogger("Mhr")
+        log.info("Received command: mhr %s (monster=%s)".format(commandName, arguments.monsterName))
+
+        respond {
+            try {
+                val monsterName = arguments.monsterName.lowercase().capitalizeWords().capitalizeDashes()
```
Comment (Yofou) [inline src/main/kotlin/extensions/mhr/Hzv.kt:40]: you don't need to manipulate this text here, as we already have the right format for the file. 
(so remove the lowercase, capitalWords and capitalizeDashes method)

File: src/main/kotlin/extensions/mhr/Hzv.kt [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+                val icon = File("$dir/monster/assets/icons/${monsterName}_Icon.PNG")
+                val hzv = File("$dir/monster/${monsterName}_HZV.PNG")
```
Comment (Yofou) [inline src/main/kotlin/extensions/mhr/Hzv.kt:42]: These file searches are case sensitive so you need  to use `.png` here all lowercase.

File: src/main/kotlin/extensions/mhr/Hzv.kt [hunk 7]
```diff
@@ -0,0 +1,62 @@
...
+
+    val dir = "./src/main/resources/source_files/MonsterDataImages/assets/mhr"
+    action {
+        val log = Logger.getLogger("Mhr")
+        log.info("Received command: mhr %s (monster=%s)".format(commandName, arguments.monsterName))
+
+        respond {
+            try {
+                val monsterName = arguments.monsterName.lowercase().capitalizeWords().capitalizeDashes()
+                val icon = File("$dir/monster/assets/icons/${monsterName}_Icon.PNG")
```
Comment (Yofou) [inline src/main/kotlin/extensions/mhr/Hzv.kt:41]: one thing here, is that there seems to be some edge cases with icons where monsterName could be `HR_{insert name here}` but the icon name is just `{insert name here}`. so can you make sure you strip out `HR_` for icons onlys

File: src/main/kotlin/extensions/mhr/Hzv.kt [hunk 8]
```diff
@@ -0,0 +1,62 @@
...
+                val hzv = File("$dir/monster/${monsterName}_HZV.PNG")
+
+                files.add(NamedFile("icon.png", icon.inputStream()))
+                files.add(NamedFile("hzv.PNG", hzv.inputStream()))
```
Comment (Yofou) [inline src/main/kotlin/extensions/mhr/Hzv.kt:45]: make sure you're only using the lowercase version of `.png`

File: src/main/kotlin/extensions/mhr/Hzv.kt [hunk 9]
```diff
@@ -0,0 +1,62 @@
...
+
+                embed {
+                    title = "__${monsterName}__"
+                    color = Color.CatBot
+
+                    image = "attachment://hzv.PNG"
+                    thumbnail {
+                        url = "attachment://icon.PNG"
```
Comment (Yofou) [inline src/main/kotlin/extensions/mhr/Hzv.kt:53]: make sure you're only using the lowercase version of `.png` inside this embed aswell

File: src/main/resources/main.js [hunk 10]
```diff
@@ -38,6 +53,27 @@ write(
...
+  for (const i in mhrIconFileObjects) {
+    item = mhrIconFileObjects[i]
+    item["title"] = item["title"].replace(" Icon", "")
+    fix[item["title"].toLowerCase().split(" ").join("")] = item
+  }
+  write(
+    './source_files/MonsterDataImages/mhr_monster_icon_map.json',
+    fix
+  );
+}
```
Comment (Yofou) [inline src/main/resources/main.js:76]: nitpick: can you make sure you call `blah` method after defining it. for better code readability

**Review by Yofou — CHANGES_REQUESTED** [review]
Looking good, defiantly an awesome start. you need make a few changes I commented below and feel free to do or ignore the nitpicks I gave!

one more thing if you look into the `main.kt` you need to add `add( ::Mhr )` inside the extensions dsl, otherwise the command won't exist.

File: src/main/kotlin/extensions/mhr/Hzv.kt (hunk 7 above)
Comment (Yofou) [reply src/main/kotlin/extensions/mhr/Hzv.kt:41]: example of this is "HR_Anjanath", `HR_Anjanath_HZV.png` exists, but `HR_Anjanath_Icon.png` does not exist. but `Anjanath_Icon.png`, **does exist**. so we should be able to use that! although correct me if I'm wrong

File: src/main/kotlin/extensions/mhr/Hzv.kt (hunk 3 above)
Comment (ricochhet) [reply src/main/kotlin/extensions/mhr/Hzv.kt:26]: yes

File: src/main/kotlin/extensions/mhr/Hzv.kt (hunk 5 above)
Comment (ricochhet) [reply src/main/kotlin/extensions/mhr/Hzv.kt:40]: Please verify this, as we use the `monsterName` variable for the embed title and the file names.

Comment (Yofou) [reply src/main/kotlin/extensions/mhr/Hzv.kt]: Yep, I kept the manipulation on the text for embed, but we did need to strip it for actual file search

**Review by Yofou — APPROVED** [review]

**Pull request merged**